include/libtorrent/session.hpp

```diff
@@ -131,6 +121,44 @@ namespace libtorrent
 		boost::shared_ptr<aux::session_impl> m_impl;
 	};
 
+	struct TORRENT_EXPORT session_params
```

Comment:
these pseudo-building/fluently pattern style is kind of controversial, I use it, but in the context of the need for a more idiomatic client code. Let me skip this in this PR (unless you think it's necessary).